lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,513 @@
+    if (!this.channel) {
+      throw errors.CONNEXT_CHANNEL_NOT_INITIATED;
+    }
+    return this.channel;
+  }
+
+  protected getTokenAddress(currency: string): string {
+    const tokenAdress = this.tokenAddresses.get(currency);
+    if (!tokenAdress) {
+      throw errors.TOKEN_ADDRESS_NOT_FOUND;
+    }
+    return tokenAdress;
+  }
+
+  protected verifyConnection = async () => {
+    this.logger.info('trying to verify connection to connext');
+    try {
+      const channel = this.getConnextChannel();
+
+      await channel.isAvailable();
+
+      this.emit('connectionVerified', { newIdentifier: channel.publicIdentifier });
+      this.setStatus(ClientStatus.ConnectionVerified);
+    } catch (err) {
+      this.logger.error(
+        `could not verify connection to connext, retrying in ${ConnextClient.RECONNECT_INTERVAL} ms`,
+        err,
+      );
+      await this.disconnect();
+    }
+  }
+
+  public sendSmallestAmount = async (
+    rHash: string,
+    destination: string,
+    currency: string,
+  ) => {
+    const tokenAddress = this.getTokenAddress(currency);
+
+    const secret = await this.executePaymentTransfer({
+      token_address: tokenAddress,
+      target_address: destination,
+      amount: 1,
+      secret_hash: rHash,
+    });
+    return secret;
+  }
+
+  public sendPayment = async (deal: SwapDeal): Promise<string> => {
+    assert(deal.state === SwapState.Active);
+    assert(deal.destination);
+    let amount: number;
+    let tokenAddress: string;
+    let lock_timeout: number | undefined;
+    if (deal.role === SwapRole.Maker) {
+      // we are the maker paying the taker
+      amount = deal.takerUnits;
+      tokenAddress = this.tokenAddresses.get(deal.takerCurrency)!;
+    } else {
+      // we are the taker paying the maker
+      amount = deal.makerUnits;
+      tokenAddress = this.tokenAddresses.get(deal.makerCurrency)!;
+      lock_timeout = deal.makerCltvDelta!;
+    }
+
+    try {
+      const secret = await this.executePaymentTransfer({
+        amount,
+        lock_timeout,
+        token_address: tokenAddress,
+        target_address: deal.destination!,
+        secret_hash: deal.rHash,
+      });
+      return secret;
+    } catch (err) {
+      switch (err.code) {
+        case 'ECONNRESET':
+        case errorCodes.UNEXPECTED:
+        case errorCodes.TIMEOUT:
+        case errorCodes.SERVER_ERROR:
+        case errorCodes.INVALID_TOKEN_PAYMENT_RESPONSE:
+          throw swapErrors.UNKNOWN_PAYMENT_ERROR(err.message);
+        default:
+          throw swapErrors.FINAL_PAYMENT_ERROR(err.message);
+      }
+    }
+  }
+
+  public addInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  public settleInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  public removeInvoice = async () => {
+    // not implemented, connext does not use invoices
+  }
+
+  public lookupPayment = async (
+    rHash: string,
```

Comment:
`rHash` is unused.

Reply:
This will be revised once we have the HTLC functionality on our client